packages/jest-cli/src/watch.js

```diff
@@ -86,7 +86,9 @@ const watch = (
       results => {
         isRunning = false;
         hasSnapshotFailure = !!results.snapshot.failure;
-        testWatcher.setState({interrupted: false});
+        if (config.bail) {
```

Comment:
Ok, seems like my fix wasn't super thorough. I need to think about this, for now using `config.bail` as a flag seems reasonable. Do you mind opening another issue linking to this piece in the code? I'll take care of it at a later time.